BotRoyaleUnity/Assets/Scripts/BuildRobotFromNetwork.cs

```diff
@@ -12,11 +12,36 @@ void Awake()
         GameStateManager.Instance.RegisterActionToState(GameStateManager.GameStates.BATTLE, socketConnection);
     }
 
+    void Start(){
+        socketIO = FindObjectOfType<SocketConnectionHandler>();
+         socketIO.OnGameMessage("changeBehaviour", jsonObject =>
+        {
+            foreach(GameObject robot in GameObject.FindGameObjectsWithTag("robot")){
+                if(robot.name == jsonObject["username"]){
+                    robot.GetComponent<RoombaMovement>().SetNavigationMode(jsonObject["behaviour"]);
+                }
+            }
+        });
+
+         socketIO.OnGameMessage("submitrobot", jsonObject =>
+        {
+            Debug.Log("Built");
```

Comment:
Im a little confused as to why this is implemented twice in this class. Seems to be a duplicate of lines 44-51

Reply:
It wasn't spawning the robots the way that it was. I had to move it to the Start() method.